0003-interledger-protocol/0003-interledger-protocol.md

```diff
@@ -172,77 +131,171 @@ As with the [internet protocol](https://tools.ietf.org/html/rfc791#section-2.3),
 
-The condition is an optional field. If no condition is provided, the funds are immediately credited to the recipient of the transfer. ("Optimistic Mode")
+2. The receiver MAY trust the sender. The sender will notify the receiver about the intent to transfer. If the receiver provides a fulfillment for the condition before the expiry date, the sender will perform a regular transfer to the receiver.
 
-#### expiresAt
+3. The sender and receiver MAY appoint a mutually trusted third-party which has an account on the local ledger. The sender performs a regular transfer into a neutral third-party account. In the first step, funds are transfered into the account belonging to the neutral third-party.
 
-    IlpExpiry :== GeneralizedTime
+### ILP Error Format
 
-Ledgers MAY require that all transfers with a condition also carry an expiry timestamp. Ledgers MUST reject transfers that carry an expiry timestamp, but no condition. Ledgers MUST reject transfers whose expiry transfer time has been reached or exceeded and whose condition has not yet been fulfilled. When rejecting a transfer, the ledger MUST lift the hold and make the funds available to the sender again.
+Here is a summary of the fields in the ILP error format:
 
-### Holds Without Native Ledger Support
+| Field | Type | Short Description |
+|:--|:--|:--|
+| code | IA5String | [ILP Error Code](#ilp-error-codes)
+| triggeredBy | Address | ILP address of the entity that originally emitted the error |
+| forwardedBy | SEQUENCE OF Address | ILP addresses of connectors that relayed the error message |
+| triggeredAt | Timestamp | Time when the error was initially emitted |
+| data | OCTET STRING | Error data provided for debugging purposes |
 
-Not all ledgers support held transfers. In the case of a ledger that doesn't, the sender and recipient of the local ledger transfer MAY choose a commonly trusted party to carry out the hold functions. There are three options:
+#### code
 
-1. The sender MAY trust the receiver. The sender will perform a regular transfer in the first step and the receiver will perform a transfer back if the condition has not been met in time.
+    IA5String (SIZE(3))
 
-2. The receiver MAY trust the sender. The sender will notify the receiver about the intent to transfer. If the receiver provides a fulfillment for the condition before the expiry date, the sender will perform a regular transfer to the receiver.
+See [ILP Error Codes](#ilp-error-codes) for the list of error codes and their meanings.
+
+#### triggeredBy
+
+[ILP Address](#account) of the entity that originally emitted the error.
+
+#### forwardedBy
+
+[ILP Addresses](#account) of the connectors that relayed the error message.
+
+#### triggeredAt
+
+    Timestamp ::= GeneralizedTime
+
+Date and time when the error was initially emitted.
+
+#### data
+
+    OCTET STRING (SIZE(0..8192))
+
+Error data provided for debugging purposes. Protocols built on top of ILP SHOULD specify the encoding format of error `data`.
```

Comment:
I don't think this is necessary because all higher level protocols will want to standardize that anyway, and there's no reason to support multiple content types at the same time.

Note that this is very similar to how there's no content type specified for the data in an ILP Payment Packet.